src/pudl_archiver/utils.py

```diff
@@ -2,13 +2,17 @@
 import asyncio
 import logging
 from collections.abc import Awaitable, Callable
-from typing import Any
+from typing import Annotated, Any
 
 import aiohttp
+from pydantic import AnyHttpUrl
+from pydantic.functional_serializers import PlainSerializer
 
 logger = logging.getLogger(f"catalystcoop.{__name__}")
 
 
+Url = Annotated[AnyHttpUrl, PlainSerializer(lambda url: str(url), return_type=str)]
```

Comment:
It seems like there's a network of changes related to URL serialization. Can you explain how it works and why it's necessary somewhere?